### PR TITLE
Pull in Web QA tests

### PR DESCRIPTION
**Pull request by stephendonner**
@jgmize and @glogiotatidis feedback?  /cc @davehunt

[WIP]; things I know need fixing:

* README

Might need changing:

* e2e-tests directory name?  Maybe just /tests/ and combine with, or fold in the current services_tests dir, combining the best of both?
* not sure if you want to submit to ActiveData[0] and/or Treeherder[1], either: https://github.com/mozmar/snippets-service/compare/mozmar:20761ac...stephendonner:8f1d271#diff-89df02f741745edd5c3fbafa86fbc26bR28

I don't know if you want *full* workflow integration, but this might be a good enough start?

[0] http://davehunt.co.uk/2017/03/21/analysing-pytest-results-using-activedata.html
[1] http://davehunt.co.uk/2017/04/10/reporting-test-results-to-treeherder.html

### CODE REVIEW

**Comment by stephendonner** [comment]
Ran against production, here:

```
sdonner-17447:e2e-tests sdonner$ export PYTEST_BASE_URL="https://snippets.mozilla.com"
sdonner-17447:e2e-tests sdonner$ tox
py27 installed: apipkg==1.4,appdirs==1.4.3,beautifulsoup4==4.6.0,blessings==1.6,execnet==1.4.1,mozlog==3.4,packaging==16.8,py==1.4.33,pyparsing==2.2.0,pytest==3.0.7,pytest-base-url==1.3.0,pytest-html==1.14.2,pytest-metadata==1.5.0,pytest-xdist==1.16.0,requests==2.14.2,six==1.10.0
py27 runtests: PYTHONHASHSEED='2206221749'
py27 runtests: commands[0] | pytest --html=results/py27.html --self-contained-html --junit-xml=results/py27.xml --log-raw=results/py27_raw.txt --log-tbpl=results/py27_tbpl.txt
================================================================== test session starts ===================================================================
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- /Users/sdonner/snippets-service/e2e-tests/.tox/py27/bin/python2.7
cachedir: .cache
metadata: {'Packages': {'pytest': '3.0.7', 'py': '1.4.33', 'pluggy': '0.4.0'}, 'Platform': 'Darwin-16.6.0-x86_64-i386-64bit', 'Python': '2.7.13', 'Plugins': {'html': '1.14.2', 'base-url': '1.3.0', 'metadata': '1.5.0', 'xdist': '1.16.0', 'mozlog': '3.4'}, 'Base URL': 'https://snippets.mozilla.com'}
baseurl: https://snippets.mozilla.com
rootdir: /Users/sdonner/snippets-service/e2e-tests, inifile: tox.ini
plugins: xdist-1.16.0, metadata-1.5.0, html-1.14.2, base-url-1.3.0, mozlog-3.4
collected 9 items 

tests/test_snippets.py::TestSnippets::test_snippet_set_present[/3/Firefox/default/default/default/en-US/release/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_snippet_set_present[/3/Firefox/default/default/default/en-US/aurora/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_snippet_set_present[/3/Firefox/default/default/default/en-US/beta/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_all_links[/3/Firefox/default/default/default/en-US/release/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_all_links[/3/Firefox/default/default/default/en-US/aurora/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_all_links[/3/Firefox/default/default/default/en-US/beta/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_that_snippets_are_well_formed_xml[/3/Firefox/default/default/default/en-US/release/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_that_snippets_are_well_formed_xml[/3/Firefox/default/default/default/en-US/aurora/default/default/default/] PASSED
tests/test_snippets.py::TestSnippets::test_that_snippets_are_well_formed_xml[/3/Firefox/default/default/default/en-US/beta/default/default/default/] PASSED

------------------------------------- generated xml file: /Users/sdonner/snippets-service/e2e-tests/results/py27.xml -------------------------------------
------------------------------------ generated html file: /Users/sdonner/snippets-service/e2e-tests/results/py27.html ------------------------------------
================================================================ 9 passed in 9.94 seconds ================================================================
flake8 installed: appdirs==1.4.3,configparser==3.5.0,enum34==1.1.6,flake8==3.3.0,mccabe==0.6.1,packaging==16.8,pycodestyle==2.3.1,pyflakes==1.5.0,pyparsing==2.2.0,six==1.10.0
flake8 runtests: PYTHONHASHSEED='2206221749'
flake8 runtests: commands[0] | flake8 .
________________________________________________________________________ summary _________________________________________________________________________
  py27: commands succeeded
  flake8: commands succeeded
  congratulations :)
```

**Comment by davehunt** [comment]
We are currently submitting these results to both ActiveData and Treeherder, so ideally this would continue. See http://treeherder.readthedocs.io/submitting_data.html#add-github-repo for how to add a GitHub repository to Treeherder, though there might be an additional step for a repo not owned by the mozilla org.

**Comment by glogiotatidis** [comment]
> e2e-tests directory name? Maybe just /tests/ and combine with, or fold in the current services_tests dir, combining the best of both?

`/tests/` sounds like a better name

File: e2e-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Snippet-tests
+=====================
+
+Automated tests for the Snippets web app
+
+Running Tests
+-------------
+
+* [Install Tox](https://tox.readthedocs.io/en/latest/install.html)
+* Run `tox`
```
Comment (glogiotatidis) [inline e2e-tests/README.md]: Add a line on how to run this against a different host to `snippets.allizom.org`

**Comment by glogiotatidis** [comment]
lgtm, just a couple of nits. thanks @stephendonner !

File: e2e-tests/README.md (hunk 1 above)
Comment (stephendonner) [inline e2e-tests/README.md]: Fixed

**Comment by stephendonner** [comment]
Should be all good now.

Here's a passing run on our Jenkins instance: https://gist.github.com/stephendonner/6caae3be836c204febcb69b21decae24 (and I re-ran it locally against staging and prod, as well).

**Comment by jgmize** [comment]
> We are currently submitting these results to both ActiveData and Treeherder, so ideally this would continue

@davehunt can you share links to views of the data in ActiveData and treeherder? I'm interested in learning more about how this is being used.

**Comment by davehunt** [comment]
@jgmize I'm working on a report from ActiveData, and should have something published automatically to S3 every hour soon. For now, a work in progress is here: https://people-mozilla.org/~dhunt/fxtest-report.html

For Treeherder, you can see the results for the snippets tests here: https://treeherder.mozilla.org/#/jobs?repo=snippets-tests. Possibly more interesting though is the addons tests, which include HTML report with videos, screenshots, etc: https://treeherder.mozilla.org/#/jobs?repo=addon-tests

**Comment by davehunt** [comment]
@jgmize we're now publishing the report to http://net-mozaws-stage-fx-test-report.s3-website-us-east-1.amazonaws.com/ once an hour.

**Comment by glogiotatidis** [comment]
Do we need `/tests/.travis.yml`?

**Comment by stephendonner** [comment]
@glogiotatidis right now that just lints our Python, but it could also be used to run the tests on PR/commit, if you'd like.

Still, for it to work, we'll need to have the mozmar account change its settings in Travis.  All of that is up to you/the team, I think!

**Comment by glogiotatidis** [comment]
> Still, for it to work, we'll need to have the mozmar account change its settings in Travis. All of that is up to you/the team, I think!

OK, we won't be using this atm since we're linting in Jenkins but let's keep it there for future reference.

**Comment by glogiotatidis** [comment]
Thanks @stephendonner

**Comment by stephendonner** [comment]
Thank you, @glogiotatidis!

**Pull request merged**